closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -941,6 +940,11 @@ boolean isRestrictedServiceAllowed(Service service) {
 
             Constraint[] constraints = providerConstraints.get(providerName);
 
+            if (constraints == null) {
+                // Continue checking if the class name is the key.
+                constraints = providerConstraints.get(providerClass);
+            }
```

Comment:
Suggest this should check `providerClassName` first.

Reply:
@keithc-ca check providerClassName first, you mean like below?

```
            Constraint[] constraints = providerConstraints.get(providerClassName);

            if (constraints == null) {
                // Continue checking if the provider name is the key.
                constraints = providerConstraints.get(providerName);
            }
```

Reply:
Yes.